### PR TITLE
Feature : InputType(entity, IRepository, Repository, Exception.InputType, documentation)

### PR DESCRIPTION
**Pull request by Ellyria34**
Done

### CODE REVIEW

File: QuikForm.Entities/InputType.cs [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
 namespace QuikForm.Entities;
-public class Input
+public class InputType
 {
+    [Key]
     public int Id { get; set; }
-    public string Type { get; set; }
+
+    [Required]
+    [MaxLength(250)]
```
Comment (htkisui) [inline QuikForm.Entities/InputType.cs]: MaxLength(255)

File: QuikForm.Entities/InputType.cs [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
 using System.Text;
 using System.Threading.Tasks;
 
 namespace QuikForm.Entities;
-public class Input
+public class InputType
 {
+    [Key]
     public int Id { get; set; }
-    public string Type { get; set; }
```
Comment (htkisui) [inline QuikForm.Entities/InputType.cs:11]: string?

File: QuikForm.Entities/InputType.cs [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
+
+    [Required]
+    [MaxLength(250)]
+    public string Name { get; set; }
```
Comment (htkisui) [inline QuikForm.Entities/InputType.cs]: string?

File: QuikForm.Entities/Question.cs [hunk 4]
```diff
@@ -17,5 +17,5 @@ public class Question
     public List<Field> Fields { get; set; } = [];
 
     public int InputId { get; set; }
-    public Input Input { get; set; }
+    public InputType InputType { get; set; }
```
Comment (htkisui) [inline QuikForm.Entities/Question.cs:20]: public InputType InputType { get; set; } = null!;

File: QuikForm.Repositories/Repositories/FormRepository.cs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+    public async Task<Form> GetByIdAsync(int id)
+    {
+        Form form = await _context.Forms.FirstOrDefaultAsync(f => f.Id == id) ?? throw new FormNotFoundException();
+        return form;
+    }
+
+    public async Task<Form> UpdateAsync(Form form)
+    {
+        Form formToUpdate = await _context.Forms.FirstOrDefaultAsync(f=>f.Id == form.Id)?? throw new FormNotFoundException();
```
Comment (htkisui) [inline QuikForm.Repositories/Repositories/FormRepository.cs]: USE CTRL-E CTRL-D!!!!!!!!!! (f=>f.Id == form.Id)

File: QuikForm.Repositories/Repositories/FormRepository.cs [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+        formToUpdate.Title = form.Title;
+        formToUpdate.Description = form.Description;
+        formToUpdate.PublishedAt = form.PublishedAt;
+        formToUpdate.ClosedAt = form.ClosedAt;
```
Comment (htkisui) [inline QuikForm.Repositories/Repositories/FormRepository.cs:55]: formToUpdate.UpdatedAt = DateTime.Now;

File: QuikForm.Repositories/Repositories/FormRepository.cs [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+{
+    private readonly ApplicationDbContext _context;
+
+    public FormRepository(ApplicationDbContext context)
+    {
+        _context = context;
+    }
+
+    public async Task CreateAsync(Form form)
+    {
```
Comment (htkisui) [inline QuikForm.Repositories/Repositories/FormRepository.cs:23]: form.CreatedAt = DateTime.Now;
form.UpdatedAt = DateTime.Now;

File: QuikForm.Repositories/Repositories/InputTypeRepository.cs [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    {
+        await _context.AddAsync(inputType);
+        await _context.SaveChangesAsync();
+    }
+
+    public async Task DeleteAsync(int id)
+    {
+        InputType inputType = await _context.InputTypes.FirstOrDefaultAsync(f => f.Id == id) ?? throw new InputTypeNotFoundException();
+        _context.InputTypes.Remove(inputType);
+        await _context.SaveChangesAsync(true);
```
Comment (htkisui) [inline QuikForm.Repositories/Repositories/InputTypeRepository.cs]: remove true

File: QuikForm.Repositories/Repositories/InputTypeRepository.cs [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    public async Task<List<InputType>> GetAllAsync()
+    {
+        return await _context.InputTypes.ToListAsync();
+    }
+
+    public async Task<InputType> GetByIdAsync(int id)
+    {
+        InputType inputType = await _context.InputTypes.FirstOrDefaultAsync(i=>i.Id == id) ?? throw new InputTypeNotFoundException();
```
Comment (htkisui) [inline QuikForm.Repositories/Repositories/InputTypeRepository.cs]: ctrl-e ctrl-d

File: QuikForm.Repositories/Repositories/FormRepository.cs [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+        await _context.AddAsync(form);
```
Comment (htkisui) [inline QuikForm.Repositories/Repositories/FormRepository.cs]: _context.Forms.AddAsync(form);

File: QuikForm.Repositories/Repositories/InputTypeRepository.cs [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+    private ApplicationDbContext _context;
+
+    public InputTypeRepository(ApplicationDbContext context)
+    {
+        _context = context;
+    }
+
+    public async Task CreateAsync(InputType inputType)
+    {
+        await _context.AddAsync(inputType);
```
Comment (htkisui) [inline QuikForm.Repositories/Repositories/InputTypeRepository.cs]: _context.InputTypes.AddAsync(inputType);

File: QuikForm.Repository.Test.Units/Repositories/FormRepositoryTests.cs [hunk 12]
```diff
@@ -0,0 +1,185 @@
...
+    #region GetByIdAsync
+    [TestMethod()]
+    public async Task GetByIdAsync_FormToSearch_FormToFound()
+    {
+        // Arrange
+        DbContextOptionsBuilder<ApplicationDbContext> builder = new DbContextOptionsBuilder<ApplicationDbContext>().UseInMemoryDatabase("QuikFormTest");
+        ApplicationDbContext context = new ApplicationDbContext(builder.Options);
+        context.Database.EnsureDeleted();
+        FormRepository formRepository = new FormRepository(context);
+        await context.AddAsync(entity: new Form { Id = 1, Title = "title1" });
```
Comment (htkisui) [inline QuikForm.Repository.Test.Units/Repositories/FormRepositoryTests.cs]: context.Forms.AddAsync(...);

File: QuikForm.Repository.Test.Units/Repositories/FormRepositoryTests.cs [hunk 13]
```diff
@@ -0,0 +1,185 @@
...
+    [ExpectedException(typeof(FormNotFoundException))]
+    public async Task DeleteAsync_FormToDelete_FormNotFound()
+    {
+        // Arrange
+        DbContextOptionsBuilder<ApplicationDbContext> builder = new DbContextOptionsBuilder<ApplicationDbContext>().UseInMemoryDatabase("QuikFormTest");
+        ApplicationDbContext context = new ApplicationDbContext(builder.Options);
+        context.Database.EnsureDeleted();
+        FormRepository formRepository = new FormRepository(context);
+
+        // Assert
```
Comment (htkisui) [inline QuikForm.Repository.Test.Units/Repositories/FormRepositoryTests.cs]: // Act

File: QuikForm.Repository.Test.Units/Repositories/FormRepositoryTests.cs [hunk 14]
```diff
@@ -0,0 +1,185 @@
...
+    [ExpectedException(typeof(FormNotFoundException))]
+    public async Task GetByIdAsync_InvalidForm_ThrowFormNotFoundException()
+    {
+        // Arrange
+        DbContextOptionsBuilder<ApplicationDbContext> builder = new DbContextOptionsBuilder<ApplicationDbContext>().UseInMemoryDatabase("QuikFormTest");
+        ApplicationDbContext context = new ApplicationDbContext(builder.Options);
+        context.Database.EnsureDeleted();
+        FormRepository formRepository = new FormRepository(context);
+
+        // Assert
```
Comment (htkisui) [inline QuikForm.Repository.Test.Units/Repositories/FormRepositoryTests.cs]: // Act

File: QuikForm.Entities/InputType.cs (hunk 2 above)
Comment (Ellyria34) [reply QuikForm.Entities/InputType.cs:11]: Type not found

File: QuikForm.Entities/Question.cs (hunk 4 above)
Comment (Ellyria34) [reply QuikForm.Entities/Question.cs:20]: ?

File: QuikForm.Repositories/Repositories/FormRepository.cs (hunk 10 above)
Comment (Ellyria34) [reply QuikForm.Repositories/Repositories/FormRepository.cs]: done

File: QuikForm.Repositories/Repositories/InputTypeRepository.cs (hunk 11 above)
Comment (Ellyria34) [reply QuikForm.Repositories/Repositories/InputTypeRepository.cs]: done

File: QuikForm.Repository.Test.Units/Repositories/FormRepositoryTests.cs (hunk 12 above)
Comment (Ellyria34) [reply QuikForm.Repository.Test.Units/Repositories/FormRepositoryTests.cs]: done

File: QuikForm.Repository.Test.Units/Repositories/FormRepositoryTests.cs (hunk 13 above)
Comment (Ellyria34) [reply QuikForm.Repository.Test.Units/Repositories/FormRepositoryTests.cs]: done

File: QuikForm.Repository.Test.Units/Repositories/FormRepositoryTests.cs (hunk 14 above)
Comment (Ellyria34) [reply QuikForm.Repository.Test.Units/Repositories/FormRepositoryTests.cs]: done

File: QuikForm.Repositories/Repositories/FormRepository.cs (hunk 6 above)
Comment (Ellyria34) [reply QuikForm.Repositories/Repositories/FormRepository.cs:55]: done

File: QuikForm.Repositories/Repositories/FormRepository.cs (hunk 7 above)
Comment (Ellyria34) [reply QuikForm.Repositories/Repositories/FormRepository.cs:23]: done

File: QuikForm.Repositories/Repositories/InputTypeRepository.cs (hunk 8 above)
Comment (Ellyria34) [reply QuikForm.Repositories/Repositories/InputTypeRepository.cs]: done

File: QuikForm.Repositories/Repositories/InputTypeRepository.cs (hunk 9 above)
Comment (Ellyria34) [reply QuikForm.Repositories/Repositories/InputTypeRepository.cs]: done

**Review by htkisui — APPROVED** [review]

**Pull request merged**